lightning/src/onion_message/messenger.rs

```diff
@@ -210,6 +219,19 @@ impl<Signer: Sign, K: Deref, L: Deref, CMH: Deref> OnionMessenger<Signer, K, L,
 		let OnionMessageContents::Custom(ref msg) = message;
 		if msg.tlv_type() < 64 { return Err(SendError::InvalidMessage) }
 
+		// If we are sending straight to a blinded route and we are the introduction node, we need to
+		// advance the blinded route by 1 hop so the second hop is the new introduction node.
+		if intermediate_nodes.len() == 0 {
+			if let Destination::BlindedRoute(ref mut blinded_route) = destination {
+				let our_node_id = self.keys_manager.get_node_id(Recipient::Node)
+					.map_err(|()| SendError::GetNodeIdFailed)?;
+				if blinded_route.introduction_node_id == our_node_id {
+					blinded_route.advance_by_one(&self.keys_manager, &self.secp_ctx)
```

Comment:
This should fail if the blinded route only contains 2 blinded hops, no :)?
Per definition we then know the receiver, which I think would motivate failing with a `TooFewBlindedHops` Err (though it will complicate the definition of that error). Currently we will fail with a `Secp256k1` error, as the packet can't be computed due to `prev_control_tlvs_ss` never being assigned when there's no unblinded path and only 1 blinded hop to compute the packet for in blinded route left. I think that `Err` is too confusing for the user and unintended.

Would be great to see test case coverage for this case as well :)!

Reply:
Ok! However in that case, the current the PR _does_ actually fail in when route contains only 2 blinded hops and we're advancing by one hop, but with an `Secp256k1` error instead. This is since our implementation can't  compute the packet with 1 blinded hop only.

Reply:
> the current PR does actually fail if route contains only 2 blinded hops and we're advancing by one hop, but with an SendError::Secp256k1 error instead

It was actually a panic 😬 fixed in the latest commit

Reply:
> It was actually a panic 😬

😬 Oh my bad! Fix looks great though :)

Reply:
Thanks for noticing it!